### PR TITLE
feat(DEV-16185): add purchase order management components and functionality

### PR DESCRIPTION
**Pull request by costa-monite**
# Purchase Orders Management

## Complete Purchase Order Lifecycle
- **Create, edit, delete, and view purchase orders**
- **Support for draft, pending, issued, and cancelled statuses**
- **Real-time PDF preview** with A4 paper format and adaptive scaling
- **Send purchase orders via email** with customizable templates
- **Line items** with price, quantity, VAT rates, and measure units

## Purchase Order Form Features
- **Vendor selection** with search and filtering
- **Configurable line items** with real-time totals calculation
- **Entity bank account and address selection**
- **Expiry date** with datepicker (primary UI) and `valid_for_days` (API field)
- **UTC-normalized date calculations** with DST-aware logic
- **Round-trip date preservation**: form → API → details view
- **VAT mode support** (inclusive/exclusive)
- **Message and footer customization**

## Tab-based Navigation
- **Payables split** into "Bills" and "Purchase Orders" tabs
- **Lazy-loaded components** for better performance
- **Context-aware "Create new" menu** adapts to active tab

# 🏗️ Shared Component Architecture

## ItemsSection Component
- **Extracted from Receivables**, now shared with Purchase Orders
- **Configuration-driven** via `ItemsSectionConfig`
- **Supports both nested** (Receivables) and **flat** (Purchase Orders) data structures
- **Real-time totals calculation**: subtotal, taxes by VAT rate, total
- **Product catalog integration** (Receivables) or **manual entry** (Purchase Orders)
- **VAT exemption rationale support**

## ConfigurableDataTable
- **Reusable table component** for Bills and Purchase Orders
- **Built on MUI DataGrid** with custom column configurations
- **Bulk actions, filtering, sorting, and pagination**

## FormErrorDisplay
- **Extracted to `ui/FormErrorDisplay`** for cross-feature reuse
- **Displays general errors** and field-specific validation messages

# 🎨 UI Components

## tab-bar.tsx
- **Animated sliding indicator** under active tab
- **Used in Payables** for Bills/Purchase Orders navigation

# 🛠️ Technical Improvements

## Currency & Price Handling
- **Enhanced price extraction** to handle string, number, and object formats
- **Currency lookup with fallback chain**: `actualCurrency → product.price.currency → item.currency`
- **Proper initialization** with `useEffect` for async-loaded settings

## Date Handling (Purchase Orders)
- **Timezone-safe UTC normalization** using `setUTCHours(0, 0, 0, 0)`
- **DST-aware calculations** using `differenceInCalendarDays` instead of `differenceInDays`
- **Proper edit mode reconstruction** from `created_at`/`issued_at` + `valid_for_days`
- **Comprehensive test coverage** with 29 passing tests for date edge cases
- **React hooks optimization** with proper `useMemo` dependencies in Overview component

## Validation with Zod
- **Purchase Order validation schema**:
  - Required fields: vendor (`counterpart_id`), line items, entity, currency
  - Line item validation: name, quantity (≥1), price (≥0), measure unit, VAT rate
  - Expiry date validation: Must be today or future date with UTC normalization
  - Cross-field validation: ensure at least one non-empty line item
  - Numeric coercion for quantity and price fields
  - Localized error messages via i18n

# Email Functionality

## Email Composition Modal
- **Migrated from mixed MUI/shadcn** to pure shadcn/ui Dialog
- **Email preview** with purchase order PDF attachment simulation
- **Default email templates** with entity and counterpart info

# API & Schema Updates

## OpenAPI Schema
- **Added Purchase Order endpoints**: GET, POST, PUT, DELETE `/payable_purchase_orders`
- **Purchase order status enum**: draft, pending, issued, cancelled
- **Line item schema** with measure units: unit, cm, day, hour, kg, litre
- **Email sending endpoint**: POST `/payable_purchase_orders/{id}/send`

# 🧪 Testing

## E2E Test Coverage
- `purchase-orders.spec.ts`
- `purchase-orders.validation.spec.ts`

## Unit Tests
- `PurchaseOrders.test.tsx`
- `PurchaseOrderForm.test.tsx`
- `purchase-order-currency.test.tsx` - Currency conversion and display logic
- `calculations-normalization.test.ts` - Date calculations with UTC normalization and DST handling

## Jest Configuration
- **Fixed CSS module transformation**: Updated `jest-css-transform.js` to handle `@import` statements in CSS files
- **All tests passing**: 8/8 PurchaseOrderForm tests, 29/29 date calculation tests

# 🔄 Refactoring

## Code Organization
- **Purchase Orders module structure**:
  ```
  PurchaseOrders/
  ├── components/          # Form components
  ├── config/             # Table configuration
  ├── hooks/              # API hooks
  ├── sections/           # Form sections
  ├── Filters/            # Filter components
  ├── utils/              # Calculation utilities
  │   ├── calculations.ts # Date & total calculations
  │   └── __tests__/     # Unit tests
  ├── __tests__/          # Integration tests
  └── ExistingPurchaseOrderDetails/  # Details view
  ```
- **Shared utilities**:
  - `ItemsSection/` - Reusable line items component
  - `ui/FormErrorDisplay/` - Error display component
  - `utils/vatUtils.ts` - VAT calculation utilities
  - `utils/calculations.ts` - Purchase order calculations with DST handling
  - `utils/currencies.ts` - Currency formatting and conversion

# 🧪 Manual QA Checklist

- [ ] **Create Purchase Order**: Fill form → save draft → verify data persisted
- [ ] **Edit Purchase Order**: Modify line items → totals update immediately
- [ ] **Expiry Date**: Pick Oct 30 → save → details shows Oct 30 → reopen shows Oct 30
- [ ] **Expiry Date (DST)**: Test dates across DST boundary (Oct 26) → calculations correct
- [ ] **Expiry Date (Timezones)**: Test in different timezones → date stays consistent
- [ ] **Email Purchase Order**: Compose email → preview → send successfully
- [ ] **Status Workflow**: Draft → Issue → verify status change
- [ ] **Tab Navigation**: Switch between Bills and Purchase Orders → data loads correctly
- [ ] **Filters**: Apply status, date range, search filters → results update
- [ ] **VAT Modes**: Toggle inclusive/exclusive → totals recalculate correctly
- [ ] **Line Items**: Add/remove/edit items → totals update in real-time
- [ ] **Currency**: Change currency → format updates throughout form
- [ ] **Permissions**: Test with different user roles → actions disabled appropriately

## Summary by CodeRabbit

* **New Features**
  * Purchase Orders added to Payables: two-tab UI, sortable/filterable table, status chip, create/edit flows, PDF viewer, preview, and delete confirmation.
  * Full Create Purchase Order experience: vendor selection, line items, VAT modes, currency handling, templates, and email composer (recipient, subject/body, preview, send).
  * “Create new” menu supports both Bills and Purchase Orders.

* **Tests**
  * Extensive unit and Playwright end-to-end coverage for Purchase Orders flows.

* **Chores**
  * Mock API fixtures/handlers, build/test tooling and template settings updates.

### CODE REVIEW

File: packages/sdk-react/config/rollup.config.mjs [hunk 1]
```diff
@@ -24,13 +24,15 @@ const rollupConfig = (packageJson, options) => {
               format: 'commonjs',
               sourcemap: true,
               interop: 'compat',
+              inlineDynamicImports: true,
```
Comment (dinis-monite) [inline packages/sdk-react/config/rollup.config.mjs:27]: Why this change?

File: packages/sdk-react/src/components/payables/PurchaseOrders/constants.ts
[the remark this answers is not in the document]
Comment (costa-monite) [reply packages/sdk-react/src/components/payables/PurchaseOrders/constants.ts:1]: no meaning behind it actually, going to rename

File: packages/sdk-react/src/components/shared/ItemsSection/ItemsSection.tsx [hunk 2]
```diff
@@ -0,0 +1,700 @@
...
+                      fieldPathSuffix === 'vat_rate_value' &&
+                      typeof value === 'number'
+                    ) {
+                      newValue = vatRateBasisPointsToPercentage(value);
+                    }
+                    setLineItemFieldValueForCurrentRow(
+                      fieldPathSuffix,
+                      newValue,
+                      options
+                    );
```
[the remark this answers is not in the document]
Comment (costa-monite) [reply packages/sdk-react/src/components/shared/ItemsSection/ItemsSection.tsx:587]: VatRateField passes the selected rate’s value in basis points (e.g., 1900, 10000).
The form model stores VAT as percentage for vat_rate_value (e.g., 19, 100).
Therefore, onLineItemValueChangeAdapter must convert basis points → percentage before storing.

File: packages/sdk-react/src/components/shared/ItemsSection/ItemsSection.tsx [hunk 3]
```diff
@@ -0,0 +1,700 @@
+import { ItemsSectionConfig } from './types';
+import { CreateProductDialog } from '@/components/receivables/InvoiceDetails/CreateReceivable/components/CreateProductDialog';
+import {
+  FormErrorDisplay,
+  useFormErrors,
+} from '@/components/receivables/InvoiceDetails/CreateReceivable/sections/components/Items/FormErrorDisplay';
```
[the remark this answers is not in the document]
Comment (costa-monite) [reply packages/sdk-react/src/components/shared/ItemsSection/ItemsSection.tsx:6]: The current import from receivables is correct for this shared component. ItemsSection uses useLineItemManagement and the receivables useFormErrors, which expect the receivables-style error shape. The PO FormErrorDisplay/useFormErrors have different types and would not match here without broader changes.

But we might agree on making the error UI overridable

File: packages/sdk-react/src/components/payables/PurchaseOrders/EmailPurchaseOrderDetails.form.components.tsx [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+export interface ControlProps {
+  subject: string;
+  body: string;
+  to: string;
+}
+
+interface RecipientSelectorProps {
+  purchaseOrderId: string;
+  control: Control<ControlProps>;
+}
```
[the remark this answers is not in the document]
Comment (costa-monite) [reply packages/sdk-react/src/components/payables/PurchaseOrders/EmailPurchaseOrderDetails.form.components.tsx:62]: it is intended (usage of non zod inferred type)

File: packages/sdk-react/src/components/payables/PurchaseOrders/sections/components/PurchaseOrderPreviewMonite.tsx [hunk 5]
```diff
@@ -0,0 +1,350 @@
...
+  const convertedLineItems =
+    line_items?.map(
+      (item): SanitizableLineItem => ({
+        ...item,
+        price: {
+          currency: item.currency as components['schemas']['CurrencyEnum'],
+          value: item.price,
+        },
+      })
+    ) || [];
```
[the remark this answers is not in the document]
Comment (costa-monite) [reply packages/sdk-react/src/components/payables/PurchaseOrders/sections/components/PurchaseOrderPreviewMonite.tsx:64]: no change needed, the price calculation and display are correct with the current implementation

File: packages/sdk-react/src/components/payables/PurchaseOrders/ExistingPurchaseOrderDetails/components/OverviewTabPanel.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import { components } from '@/api';
```
[the remark this answers is not in the document]
Comment (costa-monite) [reply packages/sdk-react/src/components/payables/PurchaseOrders/ExistingPurchaseOrderDetails/components/OverviewTabPanel.tsx:1]: the conventional way is to use "import { components } from '@/api';"

File: packages/sdk-react/src/components/payables/PurchaseOrders/validation.ts [hunk 7]
```diff
@@ -0,0 +1,228 @@
...
+        if (val === undefined || val === null || val === '') return undefined;
+
+        const num = Number(val);
+
+        return isNaN(num) ? undefined : num;
+      },
+      z.number({
+        message: t(i18n)`Price is a required field`
+      }).min(0, t(i18n)`Price must be 0 or greater`)
+    ),
```
[the remark this answers is not in the document]
Comment (costa-monite) [reply packages/sdk-react/src/components/payables/PurchaseOrders/validation.ts:56]: Actually, it is the right way to work with error messages.

File: packages/sdk-react/src/components/templateSettings/hooks/useDocumentTemplatesApi.ts [hunk 8]
```diff
@@ -19,10 +21,21 @@ export const useDocumentTemplatesApi = () => {
...
-    let filtered: components['schemas']['TemplateReceivableResponse'][] = [];
-    filtered =
+    const documentTypeMap: Record<
+      SelectableDocumentType | 'default',
+      TemplateDocumentType[]
+    > = {
+      purchase_order: ['purchase_order'],
+      receivable: ['invoice', 'receivable'],
+      default: ['invoice', 'receivable'],
+    };
```
[the remark this answers is not in the document]
Comment (costa-monite) [reply packages/sdk-react/src/components/templateSettings/hooks/useDocumentTemplatesApi.ts:31]: it is already there in the DocumentTypeEnum
